vcnc-server/vcnc-rest/src/routes/v1.js

```diff
@@ -34,51 +34,38 @@ function latency() {
 }
 
 function convertVtrqID(map) {
-  map.vtrq_id = parseInt(map.vtrq_id);
+  return Object.assign({}, map, { vtrq_id: parseInt(map.vtrq_id, 10) });
 }
 
 function convertMapVtrqID(maps) {
-  for (let i = 0; i < maps.length; i++) {
-    convertVtrqID(maps[i]);  
-  }
+  return maps.map(e => convertVtrqID(e));
 }
 
 //
 // Convert v2-format of workspace to v1 format for back compatibility
 //
 function workspace(jstr) {
-  const ws = jstr.ws;
-  const jsonWs = JSON.parse(ws);
-  const jsonMap = jsonWs.maps;
+  const jsonWs = JSON.parse(jstr.ws);
   const local = jsonWs.writeback === 'never';
-  if(jsonMap.length !== 0) {convertMapVtrqID(jsonMap);  
-    for (let i = 0; i < jsonMap.length; i++) {
-      jsonMap[i].local = local;
-    }
-  } else {
-    jsonMap = [];
-  }
-  const jsonResult = {
-    spec: jsonMap,
+  return {
+    spec:
+      convertMapVtrqID(jsonWs.maps)
+      .map(e => Object.assign({}, e, { local })),
   };
```

Comment:
If it's an empty array, yes.  

[].map(e => e) returns []

If, for example, an evil Boost property tree converted the [] into '', then it wouldn't work.  I added a check as part of PR #17